test/junit/scala/collection/immutable/LazyListTest.scala

```diff
@@ -38,46 +76,51 @@ class LazyListTest {
 
     u.head
     assertEquals(LazyListTest.serializationForceCount, 1)
 
     val data = serialize(u)
-    var i = data.indexOfSlice(from)
-    to.foreach(x => {data(i) = x; i += 1})
 
-    val ud1 = deserialize(data).asInstanceOf[LazyList[Int]]
+    val ud = deserialize(data).asInstanceOf[LazyList[Int]]
 
-    // this check failed before scala/scala#10118, deserialization triggered evaluation
     assertEquals(LazyListTest.serializationForceCount, 1)
 
-    ud1.tail.head
+    ud.tail.head
     assertEquals(LazyListTest.serializationForceCount, 2)
 
     u.tail.head
     assertEquals(LazyListTest.serializationForceCount, 3)
   }
 
+  @Test def serializationNotCompat(): Unit = {
+    import sd._
+
+    // 2.13.15 LazyList.empty[String]
+    val emptyS = "rO0ABXNyACNzY2FsYS5jb2xsZWN0aW9uLmltbXV0YWJsZS5MYXp5TGlzdAAAAAAAAAADAgAFWgAIYml0bWFwJDBaAA1taWRFdmFsdWF0aW9uWgAzc2NhbGEkY29sbGVjdGlvbiRpbW11dGFibGUkTGF6eUxpc3QkJHN0YXRlRXZhbHVhdGVkTAAJbGF6eVN0YXRldAARTHNjYWxhL0Z1bmN0aW9uMDtMACpzY2FsYSRjb2xsZWN0aW9uJGltbXV0YWJsZSRMYXp5TGlzdCQkc3RhdGV0ACtMc2NhbGEvY29sbGVjdGlvbi9pbW11dGFibGUvTGF6eUxpc3QkU3RhdGU7eHABAAFwc3IAJnNjYWxhLnJ1bnRpbWUuTW9kdWxlU2VyaWFsaXphdGlvblByb3h5AAAAAAAAAAECAAFMAAttb2R1bGVDbGFzc3QAEUxqYXZhL2xhbmcvQ2xhc3M7eHB2cgAwc2NhbGEuY29sbGVjdGlvbi5pbW11dGFibGUuTGF6eUxpc3QkU3RhdGUkRW1wdHkkAAAAAAAAAAMCAAB4cA=="
+    // 2.13.15 LazyList.from(1)
+    val from1S = "rO0ABXNyACNzY2FsYS5jb2xsZWN0aW9uLmltbXV0YWJsZS5MYXp5TGlzdAAAAAAAAAADAgAFWgAIYml0bWFwJDBaAA1taWRFdmFsdWF0aW9uWgAzc2NhbGEkY29sbGVjdGlvbiRpbW11dGFibGUkTGF6eUxpc3QkJHN0YXRlRXZhbHVhdGVkTAAJbGF6eVN0YXRldAARTHNjYWxhL0Z1bmN0aW9uMDtMACpzY2FsYSRjb2xsZWN0aW9uJGltbXV0YWJsZSRMYXp5TGlzdCQkc3RhdGV0ACtMc2NhbGEvY29sbGVjdGlvbi9pbW11dGFibGUvTGF6eUxpc3QkU3RhdGU7eHAAAABzcgAhamF2YS5sYW5nLmludm9rZS5TZXJpYWxpemVkTGFtYmRhb2HQlCwpNoUCAApJAA5pbXBsTWV0aG9kS2luZFsADGNhcHR1cmVkQXJnc3QAE1tMamF2YS9sYW5nL09iamVjdDtMAA5jYXB0dXJpbmdDbGFzc3QAEUxqYXZhL2xhbmcvQ2xhc3M7TAAYZnVuY3Rpb25hbEludGVyZmFjZUNsYXNzdAASTGphdmEvbGFuZy9TdHJpbmc7TAAdZnVuY3Rpb25hbEludGVyZmFjZU1ldGhvZE5hbWVxAH4AB0wAImZ1bmN0aW9uYWxJbnRlcmZhY2VNZXRob2RTaWduYXR1cmVxAH4AB0wACWltcGxDbGFzc3EAfgAHTAAOaW1wbE1ldGhvZE5hbWVxAH4AB0wAE2ltcGxNZXRob2RTaWduYXR1cmVxAH4AB0wAFmluc3RhbnRpYXRlZE1ldGhvZFR5cGVxAH4AB3hwAAAABnVyABNbTGphdmEubGFuZy5PYmplY3Q7kM5YnxBzKWwCAAB4cAAAAAJzcgARamF2YS5sYW5nLkludGVnZXIS4qCk94GHOAIAAUkABXZhbHVleHIAEGphdmEubGFuZy5OdW1iZXKGrJUdC5TgiwIAAHhwAAAAAXEAfgANdnIAJHNjYWxhLmNvbGxlY3Rpb24uaW1tdXRhYmxlLkxhenlMaXN0JAAAAAAAAAADAgAAeHB0AA9zY2FsYS9GdW5jdGlvbjB0AAVhcHBseXQAFCgpTGphdmEvbGFuZy9PYmplY3Q7dAAkc2NhbGEvY29sbGVjdGlvbi9pbW11dGFibGUvTGF6eUxpc3QkdAAPJGFub25mdW4kZnJvbSQydAAvKElJKUxzY2FsYS9jb2xsZWN0aW9uL2ltbXV0YWJsZS9MYXp5TGlzdCRTdGF0ZTt0AC0oKUxzY2FsYS9jb2xsZWN0aW9uL2ltbXV0YWJsZS9MYXp5TGlzdCRTdGF0ZTtw"
+    def dec(s: String) = java.util.Base64.getDecoder.decode(s)
+
+    val empty = deserialize(dec(emptyS)).asInstanceOf[LazyList[String]]
+    assert(empty.isEmpty)
+    assert(empty.toList == Nil)
+    assert(empty ne LazyList.empty[String]) // not the same instance
```

Comment:
I'm surprised this works, but... neat I guess?

Reply:
is this because there are no shared field names so the new fields end up being `null`?